### PR TITLE
isolate test env to use test specific configs

### PR DESCRIPTION
**Pull request by emmanuel-apau**
- pentest resources https://github.com/CDCgov/prime-simplereport/issues/699
- SSL policy https://github.com/CDCgov/prime-simplereport/issues/357

### CODE REVIEW

File: .github/workflows/terraform_checks.yml [hunk 1]
```diff
@@ -38,6 +38,8 @@ jobs:
           terraform fmt -check demo/persistent
           terraform fmt -check stg
           terraform fmt -check stg/persistent
+          terraform fmt -check prod
+          terraform fmt -check stg/persistent
```
Comment (jeremyzitomer-usds) [inline .github/workflows/terraform_checks.yml:42]: prod/persistent?

File: backend/src/main/resources/application-azure-test.yaml [hunk 2]
```diff
@@ -0,0 +1,19 @@
+spring:
+  profiles.include: okta-dev, auth-dev
+graphql.servlet.cors:
+  allowed-origins:
+    - https://simple-report-api-test.azurewebsites.net
+    - https://simple-report-test.azureedge.net
+    - https://test.simplereport.gov
```
Comment (benwarfield-usds) [inline backend/src/main/resources/application-azure-test.yaml:7]: Does it actually need this? I think from `/app` to `/api` the browser won't even check.

File: backend/src/main/resources/application-azure-test.yaml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  admin-emails:
```
Comment (benwarfield-usds) [inline backend/src/main/resources/application-azure-test.yaml:9]: I sort of want to factor this out into a separate property file but I think the answer is to factor it out into Okta. 😁

File: backend/src/main/resources/application-okta-dev.yaml [hunk 4]
```diff
@@ -6,16 +6,4 @@ okta:
...
-    - qom2@cdc.gov # a.k.a. abeckett@cdc.gov
-    - jonathan.eskew@cms.hhs.gov
-    - neil.s.sharma@omb.eop.gov
-    - wsullivan@cdc.gov
-    - jeremy.a.zitomer@omb.eop.gov
-    - eapau@cdc.gov
-    - qrk8@cdc.gov # Sparkle
-    - David.w.methvin@omb.eop.gov
-    - thomas.a.nielsen@omb.eop.gov
+    role-prefix: "SR-DEV-TENANT:"
```
Comment (benwarfield-usds) [inline backend/src/main/resources/application-okta-dev.yaml:9]: The battle of the terminal newlines continues, I see 😁

File: ops/test/api.tf [hunk 5]
```diff
@@ -14,16 +11,14 @@ module "simple_report_api" {
   tenant_id        = data.azurerm_client_config.current.tenant_id
 
   app_settings = {
-    SPRING_PROFILES_ACTIVE                         = "azure-dev,okta-dev,no-okta-mgmt"
+    SPRING_PROFILES_ACTIVE                         = "azure-test,okta-test,no-okta-mgmt"
```
Comment (benwarfield-usds) [inline ops/test/api.tf]: As noted, there's a conflict between the env variable and what's in the profiles.include property—I don't have a strong feeling about which should be set, but definitely only one of them should.

**Review by benwarfield-usds — APPROVED** [review]
There was one breaking issue but I think it got fixed while I was writing this comment. 😄

**Pull request merged**